src/blocks/weather/nws.rs

```diff
@@ -0,0 +1,380 @@
+pub struct Config {
+    coordinates: Option<(String, String)>,
+    #[default(12)]
+    forecast_hours: usize,
+    #[serde(default)]
+    units: UnitSystem,
+}
+
+#[derive(Clone, Debug)]
+struct LocationInfo {
+    query: String,
+    name: String,
+}
+
+pub(super) struct Service<'a> {
+    config: &'a Config,
+    location: Option<LocationInfo>,
+}
+
+impl<'a> Service<'a> {
+    pub(super) async fn new(autolocate: bool, config: &'a Config) -> Result<Service<'a>> {
+        let location = if autolocate {
+            None
+        } else {
+            let coords = config.coordinates.as_ref().error("no location given")?;
+            Some(Self::get_location_query(&coords.0, &coords.1, config.units).await?)
+        };
+        Ok(Self { config, location })
+    }
+
+    async fn get_location_query(lat: &str, lon: &str, units: UnitSystem) -> Result<LocationInfo> {
+        let points_url = format!("{API_URL}/points/{lat},{lon}");
+
+        let response: ApiPoints = REQWEST_CLIENT
+            .get(points_url)
+            .send()
+            .await
+            .error("Zone resolution request failed")?
+            .json()
+            .await
+            .error("Failed to parse zone resolution request")?;
+        let mut query = response.properties.forecast_hourly;
+        query.push_str(match units {
+            UnitSystem::Metric => "?units=si",
+            UnitSystem::Imperial => "?units=us",
+        });
+        let location = response.properties.relative_location.properties;
+        let name = format!("{}, {}", location.city, location.state);
+        Ok(LocationInfo { query, name })
+    }
+}
+
+#[derive(Deserialize, Debug)]
+struct ApiPoints {
+    properties: ApiPointsProperties,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+struct ApiPointsProperties {
+    forecast_hourly: String,
+    relative_location: ApiRelativeLocation,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+struct ApiRelativeLocation {
+    properties: ApiRelativeLocationProperties,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+struct ApiRelativeLocationProperties {
+    city: String,
+    state: String,
+}
+
+#[derive(Deserialize, Debug)]
+struct ApiForecastResponse {
+    properties: ApiForecastProperties,
+}
+
+#[derive(Deserialize, Debug)]
+struct ApiForecastProperties {
+    periods: Vec<ApiForecast>,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+struct ApiValue {
+    value: f64,
+    unit_code: String,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+struct ApiForecast {
+    is_daytime: bool,
+    temperature: ApiValue,
+    relative_humidity: ApiValue,
+    wind_speed: ApiValue,
+    wind_direction: String,
+    short_forecast: String,
+}
+
+impl ApiForecast {
+    fn wind_direction(&self) -> f64 {
+        let dir = match self.wind_direction.as_str() {
```

Comment:
I'd make this return a `Some(f64)` so that "N" is `Some(0.0)` vs "" being `None`

Reply:
Done. I changed it so that "N" will return `Some(0.0)` and any unrecognized string will return `None`.